### ISSUE TITLE
Use FancyEqMixin and FancyStrMixin for all Record_* types and Name in twisted.names.dns

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2935]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2935|
|Type|enhancement|
|Created|2007-12-05 13:53:29Z|
|Branch|https://github.com/twisted/twisted/tree/dns-str-eq-2935-3|

Many of them use one or both of these, but not all.

<details><summary>Searchable metadata</summary>

```
trac-id__2935 2935
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_dns_str_eq_2935_3 branches/dns-str-eq-2935-3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1196862809000000 1196862809000000
changetime__1214074924000000 1214074924000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

What's the status of this? It would be cool if it could land, it helps unittests a lot.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [23926]) Branching to 'dns-str-eq-2935-3'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

It was waiting for [#2944](https://github.com/twisted/twisted/issues/2944), but that was finished a while ago I guess.  Should be finished now.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @exarkun|
|-|-|

 * in EqualityTests._equalityTest, can you add a comment about why you don't use assertEquals and friends? I probably know, but it's better to have written
 * there's an extra line after _equalityTest

Everything else is great, so please merge once both points addressed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [23961]) whitespace and comment about implementation

refs [#2935](https://github.com/twisted/twisted/issues/2935)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [23962]) Merge dns-str-eq-2935-3

Author: exarkun
Reviewer: therve
Fixes: [#2935](https://github.com/twisted/twisted/issues/2935)

Define string representation and equality for all record types (using
`FancyStrMixin` and `FancyEqMixin` from `twisted.python.util`).  Also
add tests for these behaviors.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
